src/bounded/bdd.rs

```diff
@@ -179,12 +177,8 @@ fn cardinality(universe: &Universe, sort: &Sort) -> usize {
 pub fn check(
     module: &mut Module,
     universe: &Universe,
-    depth: usize,
+    depth: Option<usize>,
 ) -> Result<CheckerAnswer, CheckerError> {
-    if let Err((error, _)) = sort_check_and_infer(module) {
```

Comment:
The check function should document the precondition that the module has been typechecked, if we aren't doing it twice. Also, without this, we should be able to take an immutable reference to the module instead of a mutable one.

Reply:
Also, is there a reason to only remove this for bdd.rs? I believe it's present in the other two checkers as well.